src/macaron/slsa_analyzer/analyzer.py

```diff
@@ -167,18 +171,46 @@ def run(self, user_config: dict, skip_deps: bool = False) -> int:
             find_ctx = report.find_ctx(dup_record.pre_config.get_value("path"))
             dup_record.context = find_ctx
 
-        # Store the analysis result into the SQLite database.
-        for ctx in report.get_ctxs():
-            self.store_result_to_db(ctx)
+        # Store the analysis result(s) into the SQLite database.
+        self.db_man.create_tables()
```

Comment:
Looks like here all we need to do is to create tables for the new check's results. Instead of calling `create_tables()` to create all the tables (if they don't exist), it would make sense to have a function in the db manager that only adds the new tables for the checks results in Macaron. WDUT?

Reply:
I am just trusting SQLAlchemy to be smart enough to only create what is needed, since all the tables are known by the same MetaData object it should be fine. I probably call create_tables more often than needed. Since the tables are defined declaratively, as long as all the modules containing ORM tables have been loaded/evaluated by the time Analyzer() is initialized they should all be created there. I don't _know_ this is the case so call it here (after all checks have run) as well to be sure. The exception is the  `analyze_result` table which is declared using the Core API rather than the ORM api: https://github.com/oracle-samples/macaron/blob/ee565267d152c2e1439bec8ed96c0730d4fc8f6d/src/macaron/slsa_analyzer/analyze_context.py#L198-L198, this will need to be created later. 

> only adds the new tables for the checks results

Figuring out what tables are 'new' would be complicated, and is probably something SQLAlchemy already models somehow internally. I don't really want to try and optimise that since I don't fully understand it or see a meaningful performance impact as is.

Reply:
> Figuring out what tables are 'new' would be complicated, and is probably something SQLAlchemy already models somehow internally. I don't really want to try and optimise that since I don't fully understand it or see a meaningful performance impact as is.

We discussed about it offline. We can probably create all the tables early at load time.

Reply:
In https://github.com/oracle-samples/macaron/pull/46/commits/766a2bf3d6b1d52cd639d68e59d4013e835c655f all the other calls to create_tables are removed except for the one [here](https://github.com/oracle-samples/macaron/blob/766a2bf3d6b1d52cd639d68e59d4013e835c655f/src/macaron/slsa_analyzer/analyzer.py#L97-L97) at the end of `Analyzer`'s constructor. This is after `registry.prepare()` so all all the tables _must_ have been loaded by this point.